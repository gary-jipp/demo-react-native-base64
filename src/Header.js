import React, {useState} from 'react';
import {StyleSheet, View, Text} from "react-native";

const Header = function(props) {

  return (
    <View style={styles.header}>
      <Text style={styles.title}>Base64</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  header: {
    justifyContent: 'center',
    alignItems: 'stretch',
    height: 55,
    paddingLeft: 10,
    backgroundColor: "#2196F3"
  },
  title: {
    color: "white",
    fontSize: 20,
    fontWeight: "bold",
  }
});

export default Header;