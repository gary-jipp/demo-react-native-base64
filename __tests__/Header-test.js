import 'react-native';
import React from 'react';
import Header from 'Header';
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Header Tests', () => {

  it('Renders without crashing', () => {
    const tree = renderer.create(<Header title="Application Header"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
