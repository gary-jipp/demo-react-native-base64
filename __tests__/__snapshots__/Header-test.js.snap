// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Tests Renders and Matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#2196F3",
      "height": 55,
      "justifyContent": "center",
      "paddingLeft": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Base64
  </Text>
</View>
`;
